### PR TITLE
Adding filter to the toString() method of KnnFloatVectorQuery

### PR DESCRIPTION
**Pull request by viswanathk**
Addresses the issue #13983

### CODE REVIEW

File: lucene/core/src/test/org/apache/lucene/search/TestKnnFloatVectorQuery.java [hunk 1]
```diff
@@ -29,13 +29,7 @@
...
 import org.apache.lucene.document.KnnFloatVectorField;
 import org.apache.lucene.document.StringField;
-import org.apache.lucene.index.DirectoryReader;
-import org.apache.lucene.index.IndexReader;
-import org.apache.lucene.index.IndexWriter;
-import org.apache.lucene.index.IndexWriterConfig;
-import org.apache.lucene.index.LeafReaderContext;
-import org.apache.lucene.index.QueryTimeout;
-import org.apache.lucene.index.VectorSimilarityFunction;
+import org.apache.lucene.index.*;
```
Comment (benwtrent) [inline lucene/core/src/test/org/apache/lucene/search/TestKnnFloatVectorQuery.java:32]: Let's not collapse to `*`

File: lucene/core/src/java/org/apache/lucene/search/KnnFloatVectorQuery.java [hunk 2]
```diff
@@ -112,7 +112,14 @@ VectorScorer createVectorScorer(LeafReaderContext context, FieldInfo fi) throws
...
   @Override
   public String toString(String field) {
-    return getClass().getSimpleName() + ":" + this.field + "[" + target[0] + ",...][" + k + "]";
+    StringBuilder buffer = new StringBuilder();
+    buffer.append(getClass().getSimpleName() + ":");
+    buffer.append(this.field + "[" + target[0] + ",...]");
+    buffer.append("[" + k + "]");
+    if (this.filter != null) {
+      buffer.append(",filter:[" + this.filter + "]");
+    }
```
Comment (benwtrent) [inline lucene/core/src/java/org/apache/lucene/search/KnnFloatVectorQuery.java:121]:
```suggestion
    if (this.filter != null) {
      buffer.append("[" + this.filter + "]");
    }
```

I think that would be better.

**Review by benwtrent — CHANGES_REQUESTED** [review]
Could you update the byte knn query & DiversifyingChildern* knn queries as well?

**Comment by viswanathk** [comment]
> Could you update the byte knn query & DiversifyingChildern* knn queries as well?

I made the changes fo DiversifyingChildren*, but by byteknn do you mean `ByteVectorSimilarityQuery`?

**Comment by jpountz** [comment]
I believe that @benwtrent meant `KnnByteVectorQuery`.

**Comment by viswanathk** [comment]
I hope I got all of them now.

File: lucene/join/src/java/org/apache/lucene/search/join/DiversifyingChildrenByteKnnVectorQuery.java [hunk 3]
```diff
@@ -154,7 +154,14 @@ protected TopDocs approximateSearch(
 
   @Override
   public String toString(String field) {
-    return getClass().getSimpleName() + ":" + this.field + "[" + query[0] + ",...][" + k + "]";
```
Comment (benwtrent) [inline lucene/join/src/java/org/apache/lucene/search/join/DiversifyingChildrenByteKnnVectorQuery.java:157]: test updates for this and the float version? The tests are in: TestParentBlockJoinByteKnnVectorQuery and of course the float version as well.

Comment (viswanathk) [reply lucene/join/src/java/org/apache/lucene/search/join/DiversifyingChildrenByteKnnVectorQuery.java:157]: yeah, i did check them. they didn't have existing tests that cover `toString` so skipped them. will add that coverage though.

Comment (viswanathk) [reply lucene/join/src/java/org/apache/lucene/search/join/DiversifyingChildrenByteKnnVectorQuery.java:157]: @benwtrent for the Diversifying* does it make sense to capture the `childFilter` and `parentsFilter` in toString() as well?

Comment (jpountz) [reply lucene/join/src/java/org/apache/lucene/search/join/DiversifyingChildrenByteKnnVectorQuery.java:157]: It's not obvious to me, `parentsFilter` is more a property of the index than of the query. You could have multiple levels of parents, which would make this property relevant to print, but at the same time I don't think they have proper toString() implementations. Let's leave it out for now.

**Review by jpountz — APPROVED** [review]

**Comment by jpountz** [comment]
> Could you update the byte knn query & DiversifyingChildern* knn queries as well?

Unrelated: I noticed that the `DiversifyingChildren*` queries don't use the filter to evaluate the query, is this a bug?

**Comment by benwtrent** [comment]
> Unrelated: I noticed that the DiversifyingChildren* queries don't use the filter to evaluate the query, is this a bug?

I am unsure what you mean by this? The filter is utilized when building the accepted docs set in the super() classes.

**Review by vigyasharma — APPROVED** [review]

**Review by benwtrent — APPROVED** [review]
I think a `CHANGES` entry is in order. This seems like a nice little bug fix to aid folks in debugging issues.

@viswanathk once you add the changes entry, I can merge and backport to 10.x

**Comment by viswanathk** [comment]
> I think a `CHANGES` entry is in order. This seems like a nice little bug fix to aid folks in debugging issues.
> 
> @viswanathk once you add the changes entry, I can merge and backport to 10.x

@benwtrent Added in `CHANGES` for 10.1. Also corrected some git history mess up.

**Pull request merged**